dvc/updater.py

```diff
@@ -164,23 +179,29 @@ def _get_linux(self):
         return package_managers.get(distro.id())
 
     def _get_darwin(self):
-        if not is_binary():
-            if __file__.startswith("/usr/local/Cellar"):
-                return "formula"
-            else:
-                return "pip"
+        if is_binary():
+            return None
 
-        # NOTE: both pkg and cask put dvc binary into /usr/local/bin,
-        # so in order to know which method of installation was used,
-        # we need to actually call `brew cask`
-        ret = os.system("brew cask ls dvc")
-        if ret == 0:
-            return "cask"
+        package_manager = None
 
-        return None
+        if __file__.startswith("/usr/local/Cellar"):
+            package_manager = "formula"
+
+        dvc_path = self._get_dvc_path("darwin")
```

Comment:
would it be enough to analyze `__file__` like a couple lines above? instead of running `which`\`where`? Also, is there a more precise heuristic than `'conda' in`? I think it can easily lead to a lot of false positives - `my-conda-project/.env/bin/dvc` (installed via pip in virtualenv - the recommended way)

Reply:
But that would also mean that I'm checking only the default conda path. Also the paths end differently depending on if it is `conda` or `miniconda`. If you don't mind I try to solve this on Friday because most likely I won't find time during week for that. @shcheklein I hope you are fine with that

Reply:
@n3hrox absolutely! take your time.